Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -210,38 +211,47 @@ void PPCSymbolDB::LogFunctionCall(u32 addr)
-// only
-// a handful of functions lined up by coincidence and then you can clear the symbols. In the future
-// I
-// want to make it smarter, so it checks that there are no BLRs in the middle of the function
-// (by checking the code length), and also make it cope with added functions in the middle or work
-// based on the order of the functions and their approximate length. Currently that process has to
-// be
-// done manually and is very tedious.
-// The use case for separate handling of map files that aren't bad is that you usually want to also
-// load names that aren't functions(if included in the map file) without them being rejected as
-// invalid.
+// only a handful of functions lined up by coincidence and then you can clear the symbols. In the
+// future I want to make it smarter, so it checks that there are no BLRs in the middle of the
+// function (by checking the code length), and also make it cope with added functions in the middle
+// or work based on the order of the functions and their approximate length. Currently that process
+// has to be done manually and is very tedious. The use case for separate handling of map files that
+// aren't bad is that you usually want to also load names that aren't functions (if included in the
+// map file) without them being rejected as invalid.
 // You can see discussion about these kinds of issues here :
 // https://forums.oculus.com/viewtopic.php?f=42&t=11241&start=580
 // https://m2k2.taigaforum.com/post/metroid_prime_hacking_help_25.html#metroid_prime_hacking_help_25
 
 // This one can load both leftover map files on game discs (like Zelda), and mapfiles
 // produced by SaveSymbolMap below.
 // bad=true means carefully load map files that might not be from exactly the right version
-bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
+bool PPCSymbolDB::LoadMap(const File::Path& path, bool bad)
 {
-  File::IOFile f(filename, "r");
-  if (!f)
+  std::string contents;
+  if (!File::ReadFileToString(path, contents))
     return false;
 
   // four columns are used in American Mensa Academy map files and perhaps other games
   bool started = false, four_columns = false;
   int good_count = 0, bad_count = 0;
 
-  char line[512];
-  while (fgets(line, 512, f.GetHandle()))
+  for (size_t line_start = 0, line_end; line_start < contents.size(); line_start = line_end + 1)
   {
-    size_t length = strlen(line);
+    line_end = contents.find('\n', line_start);
+    if (line_end == std::string::npos)
+      line_end = contents.size();
+
+    std::string line_data = contents.substr(line_start, line_end - line_start);
+    size_t length = line_data.size();
+    if (length > 1 && line_data[length - 2] == '\r')
+    {
+      line_data.resize(--length);
+      line_data[length - 1] = '\n';
+    }
+
+    // This is terrible, but so is the rest of this code. I'm just trying to
+    // fit in.
+    char* line = const_cast<char*>(line_data.data());
```

Comment:
Isn't there a guaranteed null byte at the end of each std::string, meaning that `line_data[0]` always is valid to access?